File: 057-删除链表中重复的结点/057-deleteDuplication.cpp
/*
struct ListNode {
    int val;
    struct ListNode *next;
    ListNode(int x) :
        val(x), next(NULL) {
    }
};
*/
class Solution {
public:
    ListNode* deleteDuplication(ListNode* pHead)
    {
        ListNode* first=new ListNode(-1);
        first->next=pHead;
        ListNode* last=first;
        ListNode* p=pHead;
        while( p!=nullptr and p->next!=nullptr){
            if (p->next->val==p->val){
                int val=p->val;
                while(p!=nullptr and p->val==val){
                    p=p->next;
                }
                last->next=p;
                
            }
            else{
                last=p;
                p=p->next;
            }
        }
        return first->next;
    }
};

File: 057-删除链表中重复的结点/057-deleteDuplication.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    def deleteDuplication(self, pHead):
        # write code here
        first=ListNode(-1)
        first.next=pHead
        last=first
        prior=pHead
        while prior and prior.next :
            if prior.next.val==prior.val:
                val=prior.val
                while prior is not None and prior.val==val:
                    prior=prior.next
                last.next=prior
            else:
                last=prior
                prior=prior.next
        return first.next
